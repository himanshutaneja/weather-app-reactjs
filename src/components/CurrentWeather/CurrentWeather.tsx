import { FC } from 'react';
import moment from 'moment';
import { CurrentWeather as CurrentWeatherData } from '../../store/types';
import { formatTemperature } from '../../helpers';
import {
  CurrentWeatherContainer,
  Header,
  Location,
  Time,
  Detail,
  Temperature,
  Phrase,
  WeatherIcon,
  TemperatureHighLow,
} from './styles';

interface CurrentWeatherProps {
  data: CurrentWeatherData;
}

const CurrentWeather: FC<CurrentWeatherProps> = ({ data }) => (
  <CurrentWeatherContainer>
    <Header>
      <Location>
        {data.name} - {data.sys.country} Weather
      </Location>
      <Time>As of {moment.unix(data.dt).format('hh:mm A')}</Time>
    </Header>
    <Detail>
      <div>
        <Temperature>{formatTemperature(data.main.temp)}°</Temperature>
        <Phrase>{data.weather[0].description}</Phrase>
      </div>
      <div>
        <WeatherIcon
          className={`wi wi-owm-${data.weather[0].id}`}
        ></WeatherIcon>
        <TemperatureHighLow>
          {formatTemperature(data.main.temp_max)}°/
          {formatTemperature(data.main.temp_min)}°
        </TemperatureHighLow>
      </div>
    </Detail>
  </CurrentWeatherContainer>
);

export default CurrentWeather;
