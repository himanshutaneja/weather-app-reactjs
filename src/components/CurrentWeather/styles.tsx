import styled from 'styled-components';

export const CurrentWeatherContainer = styled.div`
  background-image: linear-gradient(#046f94, #3493ad);
  padding: 20px;
  border-radius: 6px;
  box-shadow: 0 0 12px 0 rgb(0 0 0 / 20%);
  margin-bottom: 20px;
`;

export const Header = styled.div`
  font-size: 1.25rem;
  font-weight: 400;
`;

export const Location = styled.div`
  font-weight: 600;
  font-size: 1.25rem;
`;

export const Time = styled.div`
  font-size: 1.125rem;
  opacity: 0.65;
`;

export const Detail = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const Temperature = styled.span`
  font-size: 6rem;
  line-height: 0.96em;
  font-weight: 500;
`;

export const Phrase = styled.div`
  font-size: 1.5rem;
  font-weight: 700;
`;

export const WeatherIcon = styled.i`
  font-size: 80px;
`;

export const TemperatureHighLow = styled.div`
  font-size: 1.5rem;
  font-weight: 700;
  text-align: right;
  margin: 12px;
`;
