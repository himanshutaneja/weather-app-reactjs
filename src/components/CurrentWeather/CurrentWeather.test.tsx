import { shallow, ShallowWrapper } from 'enzyme';
import CurrentWeather from './CurrentWeather';
import { currentWeatherMock } from '../../fixtures/CurrentWeather.fixture';

describe('CurrentWeather', () => {
  let wrapper: ShallowWrapper;
  beforeEach(() => {
    wrapper = shallow(<CurrentWeather data={currentWeatherMock} />);
  });

  it('should render search bar correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
