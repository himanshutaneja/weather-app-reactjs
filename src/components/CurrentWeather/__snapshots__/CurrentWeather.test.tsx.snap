// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentWeather should render correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      Sydney
       - 
      AU
       Weather
    </styled.div>
    <styled.div>
      As of 
      01:41 PM
    </styled.div>
  </styled.div>
  <styled.div>
    <div>
      <styled.span>
        15
        °
      </styled.span>
      <styled.div>
        broken clouds
      </styled.div>
    </div>
    <div>
      <styled.i
        className="wi wi-owm-803"
      />
      <styled.div>
        16
        °/
        14
        °
      </styled.div>
    </div>
  </styled.div>
</styled.div>
`;
